pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -97,11 +98,49 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
+	// Iterate over matrix.params and compute count of all generated combinations
+	count := m.countGeneratedCombinationsFromParams()
+
+	// Add any additional combinations generated from matrix include params
+	count += m.countNewCombinationsFromInclude()
+
+	return count
+}
+
+// countGeneratedCombinationsFromParams returns the count of combinations of Parameters generated from the matrix
+// parameters
+func (m *Matrix) countGeneratedCombinationsFromParams() int {
+	if !m.hasParams() {
+		return 0
+	}
+	count := 1
+	for _, param := range m.Params {
+		count *= len(param.Value.ArrayVal)
+	}
+	return count
+}
+
+// countNewCombinationsFromInclude returns the count of combinations of Parameters generated from the matrix
+// include parameters
+func (m *Matrix) countNewCombinationsFromInclude() int {
+	if !m.hasInclude() {
+		return 0
+	}
```

Comment:
This condition is not necessary but this is avoiding a call to `extractParamMapArrVals()` in case of no `matrix.Include` specified which is reasonable as well.